genon/preprocessor/module/utils/util.py

```
@@ -0,0 +1,146 @@
+    elif header.startswith(b"\x89PNG"):
+        return "png"
+    elif header.startswith(b"\xff\xd8\xff"):
+        return "jpg"
+
+    # 매직 헤더로 판단할 수 없으면 확장자 사용
+    return os.path.splitext(file_path)[-1].lower()
+
+
+def convert_md_to_pdf(md_path):
+    """Markdown 파일을 PDF로 변환"""
+    install_packages(["chardet"])
```

Ruff (0.15.10)

[error] Undefined name `install_packages`

(F821)